textattack/commands/augment.py

```diff
@@ -28,62 +28,101 @@ def run(self, args):
-                augmented_row = row.copy()
-                augmented_row[args.input_column] = augmentation
-                output_rows.append(augmented_row)
-        # Print to file.
-        with open(args.outfile, "w") as outfile:
-            csv_writer = csv.writer(
-                outfile, delimiter=",", quotechar='"', quoting=csv.QUOTE_MINIMAL
+            output_rows = []
+            for row in tqdm.tqdm(rows, desc="Augmenting rows"):
+                text_input = row[args.input_column]
+                if not args.exclude_original:
+                    output_rows.append(row)
+                for augmentation in augmenter.augment(text_input):
+                    augmented_row = row.copy()
+                    augmented_row[args.input_column] = augmentation
+                    output_rows.append(augmented_row)
+            # Print to file.
+            with open(args.outfile, "w") as outfile:
+                csv_writer = csv.writer(
+                    outfile, delimiter=",", quotechar='"', quoting=csv.QUOTE_MINIMAL
+                )
+                # Write header.
+                csv_writer.writerow(output_rows[0].keys())
+                # Write rows.
+                for row in output_rows:
+                    csv_writer.writerow(row.values())
+            textattack.shared.logger.info(
+                f"Wrote {len(output_rows)} augmentations to {args.outfile} in {time.time() - start_time}s."
             )
-            # Write header.
-            csv_writer.writerow(output_rows[0].keys())
-            # Write rows.
-            for row in output_rows:
-                csv_writer.writerow(row.values())
-        textattack.shared.logger.info(
-            f"Wrote {len(output_rows)} augmentations to {args.outfile} in {time.time() - start_time}s."
-        )
+        else:
+            print("\nRunning in interactive mode")
+            print("----------------------------")
+            while True:
+                print(
+                    'Enter a sentence to augment, "q" to quit, "c" to change arguments:'
+                )
+                text = input()
+
+                if text == "q":
+                    break
+
+                if text == "c":
+                    print("recipe: ", end=" ")
+                    args.recipe = input()
+                    print("pct-words-to-swap: ", end=" ")
+                    args.pct_words_to_swap = float(input())
+                    print("transformations-per-example: ", end=" ")
+                    args.transformations_per_example = int(input())
+                    continue
+
+                if not text:
+                    continue
+
+                print("Augmenting...")
+                print("----------------------------")
+
+                augmenter = eval(AUGMENTATION_RECIPE_NAMES[args.recipe])(
+                    pct_words_to_swap=args.pct_words_to_swap,
+                    transformations_per_example=args.transformations_per_example,
+                )
```

Comment:
I don't think this should happen on each iteration of the loop. Some augmenter `__init__` methods are expensive. For example, the `EmbeddingAugmenter` loads a full word embedding matrix upon initialization, which can be really slow, especially if you don't have a lot of RAM.

To avoid this, we can create the augmenter before the loop, and then re-initialize it only if the user types `"c"` (to change its arguments).